clients/client-core/src/client/real_messages_control/acknowledgement_control/action_controller.rs

```diff
@@ -246,14 +252,18 @@ impl ActionController {
     }
 
     pub(super) async fn run(&mut self) {
-        loop {
-            // at some point there will be a global shutdown signal here as the third option
+        while !self.shutdown.is_shutdown() {
             tokio::select! {
                 // we NEVER expect for ANY sender to get dropped so unwrap here is fine
```

Comment:
hehe, so when I wrote those comments about senders that couldn't be dropped I **might** had written this in the context of binary never being shutdown gracefully - perhaps double check it? (or just to be safe, match on the result and never unwrap it)